core/server/services/mega/post-email-serializer.js

```diff
@@ -4,21 +4,65 @@ const settingsCache = require('../../services/settings/cache');
 const urlUtils = require('../../lib/url-utils');
 const moment = require('moment');
 const cheerio = require('cheerio');
+const api = require('../../api');
 
 const getSite = () => {
     return Object.assign({}, settingsCache.getPublic(), {
         url: urlUtils.urlFor('home', true)
     });
 };
 
-const serialize = (post) => {
+/**
+ * createUnsubscribeUrl
+ *
+ * Takes a member and returns the url that should be used to unsubscribe
+ * In case of no member, generates the preview unsubscribe url - `?preview=1`
+ *
+ * @param {object} member
+ * @param {string} member.uuid
+ */
+const createUnsubscribeUrl = (member) => {
+    const siteUrl = urlUtils.getSiteUrl();
+    const unsubscribeUrl = new URL(siteUrl);
+    unsubscribeUrl.pathname = `${unsubscribeUrl.pathname}/unsubscribe/`.replace('//', '/');
+    if (member.uuid) {
+        unsubscribeUrl.searchParams.set('uuid', member.uuid);
+    } else {
+        unsubscribeUrl.searchParams.set('preview', '1');
+    }
+
+    return unsubscribeUrl.href;
+};
+
+// NOTE: serialization is needed to make sure we are using current API and do post transformations
+//       such as image URL transformation from relative to absolute
+const serializePostModel = async (model) => {
+    const frame = {options: {context: {user: true}}};
+    const apiVersion = model.get('api_version') || 'v3';
+    const docName = 'posts';
+
+    await api.shared
+        .serializers
+        .handle
+        .output(model, {docName: docName, method: 'read'}, api[apiVersion].serializers.output, frame);
+
+    return frame.response[docName][0];
+};
+
+const serialize = async (postModel, options = {isBrowserPreview: false}) => {
+    const post = await serializePostModel(postModel);
     post.published_at = post.published_at ? moment(post.published_at).format('DD MMM YYYY') : moment().format('DD MMM YYYY');
     post.authors = post.authors && post.authors.map(author => author.name).join(',');
     post.html = post.html || '';
     if (post.posts_meta) {
         post.email_subject = post.posts_meta.email_subject;
     }
-    let juicedHtml = juice(template({post, site: getSite()}));
+    let htmlTemplate = template({post, site: getSite()});
+    if (options.isBrowserPreview) {
+        const previewUnsubscribeUrl = createUnsubscribeUrl({});
```

Comment:
@rishabhgrg would not need to pass this empty object if small refactor applied from above comment ;)